### PR TITLE
Tree intersection

### PR DESCRIPTION
**Pull request by kyllo34**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code-challenges/treeIntersection/tree-intersection.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const allMap = new Map()
+  const collisionsMap = new Map()
+  preOrder(tree1.root, allMap, collisionsMap)
+  preOrder(tree2.root, allMap, collisionsMap)
+  return [...collisionsMap.keys()];
+}
+function preOrder(current, allMap, collisionsMap) {
+  // console.log(current)
+  if (!current) return
+  //if the map already has the value, add the value to the collisions map.
```
Comment (XiaojuanCai) [inline code-challenges/treeIntersection/tree-intersection.test.js]: Andrew, 
think about the case that tree1 has duplicate values. Will your program add the duplicate values to the collisionMap?
For example, tree1 is 1->1, tree2 is null. What's your output?
